runtime/permissions/lib.rs

```diff
@@ -950,6 +949,15 @@ impl NetDescriptor {
 
     Ok(NetDescriptor(host, port))
   }
+
+  pub fn from_url(url: &Url) -> Result<Self, AnyError> {
```

Comment:
Will this one work correctly for IP addresses?

Reply:
Seems to based on the tests.